pkg/controller/hibernation/gcp_actuator.go

```diff
@@ -0,0 +1,168 @@
+package hibernation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	compute "google.golang.org/api/compute/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	"github.com/openshift/hive/pkg/gcpclient"
+)
+
+const (
+	instanceFields = "items/*/instances(name,zone,status),nextPageToken"
+)
+
+var (
+	gcpRunningStatuses           = sets.NewString("RUNNING")
+	gcpStoppedStatuses           = sets.NewString("STOPPED", "TERMINATED")
+	gcpPendingStatuses           = sets.NewString("PROVISIONING", "STAGING")
+	gcpStoppingStatuses          = sets.NewString("STOPPING")
+	gcpRunningOrPendingStatuses  = gcpRunningStatuses.Union(gcpPendingStatuses)
+	gcpStoppedOrStoppingStatuses = gcpStoppedStatuses.Union(gcpStoppingStatuses)
+	gcpNotRunningStatuses        = gcpStoppedOrStoppingStatuses.Union(gcpPendingStatuses)
+	gcpNotStoppedStatuses        = gcpRunningOrPendingStatuses.Union(gcpStoppingStatuses)
+)
+
+func init() {
+	RegisterActuator(&gcpActuator{getGCPClientFn: getGCPClient})
+}
+
+type gcpActuator struct {
+	getGCPClientFn func(*hivev1.ClusterDeployment, client.Client, log.FieldLogger) (gcpclient.Client, error)
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeployment
+func (a *gcpActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
+	return cd.Spec.Platform.GCP != nil
+}
+
+// StopMachines will start machines belonging to the given ClusterDeployment
+func (a *gcpActuator) StopMachines(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) error {
+	logger = logger.WithField("cloud", "GCP")
+	gcpClient, err := a.getGCPClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return err
+	}
+	instances, err := gcpClient.ListComputeInstances(gcpclient.ListComputeInstancesOptions{
+		Filter: instanceFilter(cd),
+		Fields: instanceFields,
+	})
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "Failed to fetch instances")
+		return errors.Wrap(err, "failed to fetch instances")
+	}
+	instances = filterByStatus(instances, gcpRunningOrPendingStatuses)
+	errs := []error{}
```

Comment:
To save an allocation when there are no errors.
```suggestion
	var errs []error
```